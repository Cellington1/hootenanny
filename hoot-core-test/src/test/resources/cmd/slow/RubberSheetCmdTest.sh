#!/bin/bash
set -e

mkdir -p $HOOT_HOME/test-output/cmd/slow/RubberSheetCmdTest

input1=$HOOT_HOME/test-files/DcGisRoads.osm
input2=$HOOT_HOME/test-files/DcTigerRoads.osm
outputfile=$HOOT_HOME/test-output/cmd/slow/RubberSheetCmdTest/out.osm
comparefile=$HOOT_HOME/hoot-core-test/src/test/resources/cmd/slow/RubberSheetCmdTest/out.osm

# Run the command. Note that we sanitize output to remove timestamps
hoot rubber-sheet $input1 $input2 $outputfile | \
           sed "s/^[0-9]\{2\}:[0-9]\{2\}:[0-9]\{2\}\.[0-9]\{3\} //g" |\
           sed "s/RMSE: [0-9]\+\.[0-9]\+//g"

# Check output
hoot is-match $comparefile $outputfile || diff $comparefile $outputfile

