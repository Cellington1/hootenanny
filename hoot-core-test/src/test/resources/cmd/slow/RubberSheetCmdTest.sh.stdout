INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::ReprojectToPlanarOp
Reprojecting 0 / 7487         Reprojecting 1000 / 7487         Reprojecting 2000 / 7487         Reprojecting 3000 / 7487         Reprojecting 4000 / 7487         Reprojecting 5000 / 7487         Reprojecting 6000 / 7487         Reprojecting 7000 / 7487         
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DuplicateWayRemover
INFO  ...nflate/DuplicateWayRemover.cpp(  64) Removing duplicate ways...
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SuperfluousWayRemover
INFO  ...late/SuperfluousWayRemover.cpp(  59) Removing superfluous ways...
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::IntersectionSplitter
INFO  ...itter/IntersectionSplitter.cpp( 109) Splitting intersections...
  Intersection splitter todo: 0       
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::UnlikelyIntersectionRemover
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DualWaySplitter
INFO  ...e/splitter/DualWaySplitter.cpp(  83) Assuming drives on right.
INFO  ...e/splitter/DualWaySplitter.cpp( 246) Splitting divided ways...

INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::ImpliedDividedMarker
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DuplicateNameRemover
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SmallWayMerger
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  74) Applying visitor: hoot::RemoveEmptyAreasVisitor
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  74) Applying visitor: hoot::RemoveDuplicateAreaVisitor
INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::NoInformationElementRemover
INFO  .../core/conflate/RubberSheet.cpp( 325) Found 237 potential ties.
INFO  .../core/conflate/RubberSheet.cpp( 363) Found 173 confident ties.
INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: DelaunayInterpolator 
INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: IDW, p: 0 
INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: Kernel estimation, sigma: 3176.88 
INFO  .../core/conflate/RubberSheet.cpp( 258) Best candidate: Kernel estimation, sigma: 3176.88
INFO  .../core/conflate/RubberSheet.cpp( 372) Kernel estimation, sigma: 3176.88
INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: Kernel estimation, sigma: 3177.26 
INFO  .../core/conflate/RubberSheet.cpp( 258) Best candidate: Kernel estimation, sigma: 3177.26
INFO  .../core/conflate/RubberSheet.cpp( 376) Kernel estimation, sigma: 3177.26
Reprojecting 0 / 7500         Reprojecting 1000 / 7500         Reprojecting 2000 / 7500         Reprojecting 3000 / 7500         Reprojecting 4000 / 7500         Reprojecting 5000 / 7500         Reprojecting 6000 / 7500         Reprojecting 7000 / 7500         
